File: vol1/ch7/LeetCode701.cpp
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        TreeNode*s=new TreeNode(val);
        if(root==NULL) return s;
        if(val<root->val) root->left=insertIntoBST(root->left,val);
        else root->right=insertIntoBST(root->right,val);
        return root;
    }
};

File: vol1/ch7/LeetCode701.py
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        s=TreeNode(val)
        if root==None:return s
        if val<root.val:root.left=self.insertIntoBST(root.left,val)
        else: root.right=self.insertIntoBST(root.right,val)
        return root
